R/capacity/health-inequalities/england/access-availability/ae-waiting-times.R

```diff
@@ -65,28 +56,207 @@ ae_double <-
-# matter that there were some NA's in the above steps that get dropped? Why is
-# this data not available?
+change_in_wait_prop |>
+  ggplot(aes(x = change)) +
+  geom_boxplot()
+
+mising_trusts_estimates <- not_reporting_trusts |>
+  left_join(ae_april_2019, by = "trust_code") |>
+  mutate(old_wait_prop = ae_over_4_hours_wait / ae_total_wait) |>
+  mutate(updated_prop = old_wait_prop + median(change_in_wait_prop$change, na.rm = T)) |>
+  select(trust_code, updated_prop)
+# Have checked trust changes and can't find the previous trust codes for R0D (i.e. RD3 & RDZ) in the 2019 data
+# Will impute using the median waiting prop for all of the data
+
+ae_double_updated <- ae_double |>
+  left_join(mising_trusts_estimates) |>
+  mutate(ae_over_4_hours_wait = ifelse(!is.na(updated_prop), ae_total_wait * updated_prop, ae_over_4_hours_wait)) |>
+  mutate(over_4_hours_wait_prop = ae_over_4_hours_wait / ae_total_wait) |>
+  mutate(ae_over_4_hours_wait = ifelse(trust_code == "R0D", ae_total_wait * median(over_4_hours_wait_prop, na.rm = T), ae_over_4_hours_wait)) |>
+  select(-updated_prop, -over_4_hours_wait_prop)
+
+
+# Join trust to LAD lookup --------
+
+lookup_trust_lad <- read_rds("data/lookup_trust_lad.rds")
+
+# Trust to LAD table only has data for acute trusts
+open_trusts |>
+  left_join(ae_double_updated) |>
+  left_join(lookup_trust_lad) |>
+  group_by(primary_category) |>
+  summarise(count = n(), prop_with_lookup = sum(!is.na(lad_code)) / n())
+
+# Current approach is to drop information on non-acute trusts since can't proportion these to MSOA
+# For the acute trusts data proportion these to LAD and normalise by total A&E count
+ae_wait_joined <- open_trusts |>
+  left_join(ae_double_updated) |>
+  inner_join(lookup_trust_lad)
+
+# Check missings
+ae_wait_joined |>
+  distinct(trust_code, primary_category, ae_over_4_hours_wait) |>
+  group_by(primary_category) |>
+  summarise(count = n(), prop_missing = sum(is.na(ae_over_4_hours_wait)) / n())
+# 5 trusts with no A&E data
+
+# Some of the open trusts don't have the A&E waiting data
+ae_wait_joined |>
+  filter(is.na(ae_over_4_hours_wait)) |>
+  distinct(trust_code, primary_category, ae_over_4_hours_wait, ae_total_wait) |>
+  left_join(points_nhs_trusts, by = c("trust_code" = "nhs_trust_code"))
+# Are specialist and may not have an A&E facility
+
+# Check highest % of a LAD missing info due to these Trusts with no A&E data
+ae_wait_joined |>
+  mutate(prop_missing = is.na(ae_over_4_hours_wait) * trust_prop_by_lad) |>
+  group_by(lad_code) |>
+  summarise(prop_missing = sum(prop_missing)) |>
+  arrange(desc(prop_missing))
+# 7 LADs between 10 - 50% missing, rest below 10%
+
+# need to re-weight the trust_prop_to_lad due to Trusts which don't have A&E
+ae_wait_lad <- ae_wait_joined |>
+  filter(!is.na(ae_over_4_hours_wait)) |>
+  group_by(lad_code) |>
+  mutate(trust_prop_by_lad_reweight = trust_prop_by_lad / sum(trust_prop_by_lad)) |>
+  ungroup() |>
```

Comment:
Sorry I think I was testing some things out and left that in and it isn't used (perhaps keeping in case ever needed to map from LAD to Trust, but unlikely). Will remove to avoid confusion.

Reply:
Removed in https://github.com/britishredcrosssociety/resilience-index/pull/6/commits/27d5b36d9e2c112e3d2b23a9dda2065e3910c828.